src/Microsoft.DotNet.Build.Tasks.Packaging/src/CreateTrimDependencyGroups.cs

```diff
@@ -65,127 +68,84 @@ public override bool Execute()
 
             var index = PackageIndex.Load(PackageIndexes.Select(pi => pi.GetMetadata("FullPath")));
 
-            // Retrieve the list of generation dependency group TFM's
-            var dependencyGroups = Dependencies.GroupBy(d => d.GetMetadata("TargetFramework")).Select(dg => new
-                {
-                    Framework = NuGetFramework.Parse(dg.Key),
-                    Dependencies = dg.ToArray()
-                });
-
-            List<ITaskItem> addedDependencies = new List<ITaskItem>();
-
-            // Exclude any non-portable frameworks that already have specific dependency groups.
-            var frameworksToExclude = dependencyGroups.Select(dg => dg.Framework).Where(fx => !FrameworkUtilities.IsGenerationMoniker(fx));
+            // Retrieve the list of dependency group TFM's
+            var dependencyGroups = Dependencies
+                .Select(i => new TaskItemPackageDependency(i))
```

Comment:
NIT: perhaps choosing better linq variables would make this line a bit more readable? for example using `dependencies` instead of `i`

Reply:
sure I can fix that.